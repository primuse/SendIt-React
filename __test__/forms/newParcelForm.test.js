import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import NewParcelForm, { NewParcelModalForm }
  from '../../src/components/forms/newParcelForm.jsx';

const onChange = jest.fn();
const createNewParcel = jest.fn();
const props = {
  parcel: {
    isLoading: false
  }
};
const mockStore = configureStore([thunk]);
const store = mockStore(props);
let wrapper;

describe('New parcel form', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}>
        <NewParcelForm />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('New parcel form', () => {
  beforeEach(() => {
    wrapper = mount(<NewParcelModalForm {...props}
      onChange={onChange} createNewParcel = { createNewParcel} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should change state', () => {
    const input = wrapper.find('#parcelName');
    input.instance().value = 'Rice';
    input.simulate('change');
    expect(wrapper.state().parcelName).toEqual('Rice');
  });
  it('should simulate the onSubmit', () => {
    wrapper.find('form').simulate('submit');
    const input1 = wrapper.find('input').at(0);
    const input2 = wrapper.find('input').at(1);
    const textArea = wrapper.find('textarea');
    const input3 = wrapper.find('input').at(2);
    const input4 = wrapper.find('input').at(3);
    const input5 = wrapper.find('input').at(4);
    const input6 = wrapper.find('input').at(5);
    input1.simulate('change', { target: { value: 'Rice' } });
    input2.simulate('change', { target: { value: '10' } });
    textArea.simulate('change', { target: { value: 'Owerri' } });
    input3.simulate('change', { target: { value: 'yinks' } });
    input4.simulate('change', { target: { value: 'yinks@gmail.com' } });
    input5.simulate('change', { target: { value: '08129844567' } });
    input6.simulate('change', { target: { value: 'Yaba' } });
    expect(createNewParcel).toHaveBeenCalled();
  });
});
