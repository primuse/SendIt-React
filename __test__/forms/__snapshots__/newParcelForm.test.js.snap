// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New parcel form renders correctly 1`] = `
<form
  className="text-right"
  id="createParcel"
  onSubmit={[Function]}
>
  <div
    className="column is-two htmlForm-row text-left"
  >
    <div>
      <label
        htmlFor="parcelType"
      >
        Parcel Name
      </label>
      <input
        id="parcelName"
        name="parcelName"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="weight"
      >
        Weight
      </label>
      <input
        id="weight"
        name="weight"
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
    </div>
  </div>
  <div
    className="htmlForm-row text-left"
  >
    <div>
      <label
        htmlFor="destination"
      >
        Destination
      </label>
      <textarea
        id="destination"
        name="destination"
        onChange={[Function]}
        required={true}
        value=""
      />
    </div>
  </div>
  <h5
    className="text-center"
  >
    Addressed To
  </h5>
  <div
    className="column is-two htmlForm-row text-left"
  >
    <div>
      <label
        htmlFor="name"
      >
        Receiver
      </label>
      <input
        id="receiver"
        name="receiver"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="column is-two htmlForm-row text-left"
  >
    <div>
      <label
        htmlFor="phoneNumber"
      >
        PhoneNumber
      </label>
      <input
        id="phoneNumber"
        name="phoneNumber"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-14"
    >
      <label
        htmlFor="PickupLocation"
      >
        Pickup Location
      </label>
      <input
        id="PickupLocation"
        name="pickupLocation"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="btn sm bg-bright-blue text-right white"
    type="submit"
  >
    Create
  </button>
</form>
`;
